internal/controller/postgrescluster/patroni_test.go

```diff
@@ -146,10 +160,47 @@ ownerReferences:
 
 			service, err := reconciler.generatePatroniLeaderLeaseService(cluster)
 			assert.NilError(t, err)
-			alwaysExpect(t, service)
+			alwaysExpect(t, service, false)
 			test.Expect(t, service)
 		})
 	}
+
+	typesAndPort := []struct {
+		Description string
+		Type        string
+		NodePort    *int32
+		Expect      func(testing.TB, *corev1.Service)
+	}{
+		{Description: "ClusterIP with Port 32000", Type: "ClusterIP",
+			NodePort: initialize.Int32(32000), Expect: func(t testing.TB, service *corev1.Service) {
+				assert.Equal(t, service.Spec.Type, corev1.ServiceTypeClusterIP)
```

Comment:
🔧 This is never called.

🤔 Perhaps the expectation should be `Expect      func(testing.TB, *corev1.Service, error)` then these cases show when there's an error or not.

```suggestion
				assert.Assert(t, service == nil)
				assert.ErrorContains(t, err, "NodePort cannot be set with type ClusterIP on pg2-ha")
```

The other cases can do `alwaysExpect(t, service)` and assert the ports here, too.